Mathlib/Computability/NFA.lean

```diff
@@ -88,12 +88,20 @@ theorem eval_singleton (a : α) : M.eval [a] = M.stepSet M.start a :=
 theorem eval_append_singleton (x : List α) (a : α) : M.eval (x ++ [a]) = M.stepSet (M.eval x) a :=
   evalFrom_append_singleton _ _ _ _
 
+/-- `M.acceptsFrom S` is the language of `x` such that `M.evalFrom S x` is an accept state. -/
+def acceptsFrom (S : Set σ) : Language α := {x | ∃ s ∈ M.accept, s ∈ M.evalFrom S x}
+
+theorem mem_acceptsFrom {S : Set σ} {x : List α} :
+    x ∈ M.acceptsFrom S ↔ ∃ s ∈ M.accept, s ∈ M.evalFrom S x := Iff.rfl
+
 /-- `M.accepts` is the language of `x` such that there is an accept state in `M.eval x`. -/
 def accepts : Language α := {x | ∃ S ∈ M.accept, S ∈ M.eval x}
 
 theorem mem_accepts {x : List α} : x ∈ M.accepts ↔ ∃ S ∈ M.accept, S ∈ M.evalFrom M.start x := by
   rfl
 
+lemma accepts_acceptsFrom : M.accepts = M.acceptsFrom M.start := rfl
+
```

Comment:
Now that `acceptsFrom` exists, `accepts` can be simply defined as `M.acceptsFrom M.start`. This is how it's done in DFA, c.f. 73756969.

Reply:
Yes, true. Admittedly, I am not overly fond of potentially repairing other proofs in the file if I replace this definition... :grimacing: 

But maybe it's work the change I'm not sure.

Reply:
That's the thing: so far, everything uses `mem_accepts` instead of unfolding the definition, and because the equality between the old and new definitions holds by reflexivity, the set of def-eqs also doesn't change.

Reply:
Ah, good point, my bad. I followed your suggestion.